mevislab.github.io/content/tutorials/summary/summary3.md

```diff
@@ -0,0 +1,691 @@
+      layout = Horizontal
+      Viewer View2D.self {
+        expandX = True
+        expandY = True
+        type    = SoRenderArea
+      }
+      Viewer SoExaminerViewer.self {
+        expandX = True
+        expandY = True
+        type    = SoExaminerViewer
+      }
+    }
+    Box Settings {
+      layout = Horizontal
+      Field selectOverlayColor {
+        title   = Color
+      }
+      Field imageAlpha {
+        step    = 0.1
+        slider  = True
+      }
+      Field thresholdInterval {
+        step    = 0.1
+        slider  = True
+      }
+      Field isoValueImage {
+        step    = 2
+        slider  = True
+      }
+    }
+    Box Info {
+      layout    = Horizontal
+    }
+  }
+}
+
+```
+{{</highlight>}}
+
+Your User Interface of the Macro Module should now look similar to this:
+
+![User Interface without Python Scripting](/images/tutorials/summary/Example3_4.png "User Interface without Python Scripting")
+
+For the next elements, we require Python scripting. Nevertheless, you are already able to use your application and perform the basic functionalities without writing any line of code.
+
+### Python scripting
+Python scripting is always necessary in case you do not want to re-use an existing Field for your User Interface but implement functions to define what happens in case of any event.
+
+Events can be raised by the user (i.e. by clicking a Button) or by the application itself (i.e. when the Window is opened).
+
+#### 3D visualization selection
+You will now add a selection possibility for the 3D viewer. This allows you to define the visibility of the 3D objects Image, Segmentation or Both.
+
+Add another *Field* to your *Parameters* section. Define the *Field* as *selected3DView* and set *type = Enum* and *values =Segmentation,Image,Both*.
+
+Add a *ComboBox* to your *Settings* and use the *Field* name defined above. Set *alignX = Left* and *editable = False* and open the *Window* of the Macro Module in MeVisLab.
+
+The values of the Field can be selected, but nothing happens in our viewers. We need to implement a *FieldListener* in Python which reacts on any value changes of the *Field selected3DView*.
+
+Open your script file and go to the *Commands* section. Add a *FieldListener* and re-use the name of our internal *Field selected3DView*. Add a *Command* to the *FieldListener* calling a Python function *viewSelectionChanged*.
+
+{{< highlight filename="<MACRO_NAME>.script" >}}
+```Stan
+Commands {
+  source = $(LOCAL)/TutorialSummary.py
+  
+  FieldListener selected3DView {
+    command = viewSelectionChanged
+  }
+}
+```
+{{</highlight>}}
+
+Right-click {{< mousebutton "right" >}} the command and select {{< menuitem "Create Python Function 'viewSelectionChanged'" >}}. MATE automatically opens the Python file of your Macro Module and creates a function *viewSelectionChanged*.
+
+{{< highlight filename="<MACRO_NAME>.py" >}}
+```Python
+from mevis import *
+
+def viewSelectionChanged(field):
```

Comment:
It would probably be better to use 4 instead of 2 spaces thoughout in any Python code, as defined by the Python formatting guidelines (PEP8).

Reply:
large effort because i need to change all python code parts then.